qiskit/extensions/standard/rz.py

```diff
@@ -48,25 +49,38 @@ def inverse(self):
         """
         return RZGate(-self.params[0])
 
+    def to_matrix(self):
```

Comment:
For now, to_matrix on RZGate should be left out (because it differs by a global phase from its definition in terms of U1, which is causing some test failures). This problem is fixed by #3472 so we can add the to_matrix (and its definition in the docs) there.

Reply:
Or if you wanted to add the parameterized matrix definition to the docs statically, you could take it from that PR:

```
    .. math::

        U_{\text{RZ}}(\theta)
            = \exp\left(-i \frac{\theta}{2} \sigma_Z \right)
            = \begin{bmatrix}
                e^{-i \theta/2} & 0 \\
                0 & e^{i \theta/2}
            \end{bmatrix}
```

Reply:
Removed the to_matrix.